### PR TITLE
Merging functionality upstream.

### PR DESCRIPTION
**Pull request by GearsAD**
Hi Glesica,

Would this PR be good? It brings back some of your functionality from the earlier version of the library.

Thanks,
Sam

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,4 +9,4 @@ before_install:
   - tar -xzf neo4j-community-2.3.4-unix.tar.gz
   - sed -i 's/#dbms.security.auth_enabled=false/dbms.security.auth_enabled=false/g' neo4j-community-2.3.4/conf/neo4j-server.properties
   - sed -i 's/dbms.security.auth_enabled=true/dbms.security.auth_enabled=false/g' neo4j-community-2.3.4/conf/neo4j-server.properties
-  - neo4j-community-2.3.4/bin/neo4j start
+- neo4j-community-2.3.4/bin/neo4j start
```
Comment (dehann) [inline .travis.yml]: @GearsAD , not sure you should delete this space

**Review by dehann — APPROVED** [review]
fixed .travis.yml, travis no passes test

**Comment by glesica** [comment]
Looks good, I'll go ahead and merge this unless there's anything else you'd like to add or change.

File: src/Neo4j.jl [hunk 2]
```diff
@@ -46,6 +59,87 @@ function connheaders(c::Connection)
...
+    Node(data::JSONObject, graph::Graph) = new(data["paged_traverse"], data["labels"],
+         data["outgoing_relationships"], data["traverse"], data["all_typed_relationships"],
+         data["all_relationships"], data["property"],
+         data["self"], data["outgoing_typed_relationships"], data["properties"],
+         data["incoming_relationships"], data["incoming_typed_relationships"],
+         data["create_relationship"], data["data"], data["metadata"],
+         split(data["self"], "/")[end] |> parse, graph)
+end
+
+function convert(::Type{Union{Dict{AbstractString,Any},Void}}, d::Dict{AbstractString,Any}) #UTF8String
```
Comment (tkelman) [inline src/Neo4j.jl:139]: please do not define this method, this changes the behavior of unrelated code
